README.mdx

```diff
@@ -0,0 +1,206 @@
+
+## Setup account & Mint KittyVerse NFT
+
+- send `tx_01_setup_kittyverse_account.cdc` signing with `0x02`
+- send `tx_02_mint_kittyverse_nft.cdc` signing with `0x02`
+- send `tx_02_mint_kittyverse_nft.cdc` again signing with `0x02`
+- query NFT.id - execute `script_01_get_kittyverse_ids.cdc` providing `0x02` as argument
+    - see two unique NFT IDS - take note of them, next let's prep them some hats
+
+# Define our Attachments - KittyHats
+
+## Attachments
+
+- What are they?
+    - In Cadence
+    - Simple mental model
+        - "Like glue between resources"
+        - ![KittyVerse & KittyHats attached](./diagrams/cat_with_hat.png)
+- Why do they matter?
+    - Can create attachments for anything without needing original developer's:
+        - Participation
+        - Permission
+
+### Overview
+
+- Conceptual
+    - Attachments are a special type that can be added to resources (doc link needed)
+    - You might be tempted to simply define all your functionality in the attachment, but it's important to know that
+      attachments are not resources.
+        - You'll lose the existence and uniqueness guarantees inherent to resources as well as make access control
+          significantly harder for yourself
+        - As mentioned earlier, attachments are best thought of as glue between resources
+            - Define an attachment that can hold some resource, designed to be added to a base resource
+    - Note that you need access to the base resource to add an attachment to it
+        - So to add an attachment to an NFT, you need to first withdraw that NFT from its Collection
+- Syntax
+    - Accessing an attachment
+    ```cadence
+    let attachmentRef: &KittyHats.HatAttachment? = nft[KittyHats.HatAttachment]
+    ```
+    - Adding an attachment
+    ```cadence
+    access(self) fun addAttachment(toNFT: @KittyVerse.NFT): @KittyVerse.NFT {
+        // If attachment already exists, return
+        if toNFT[HatAttachment] != nil {
+            return <-toNFT
+        }
+        // Otherwise, add the attachment
+        return <- attach HatAttachment() to <- toNFT
+    }
+    // Otherwise, add the attachment
+    return <- attach HatAttachment() to <- toNFT
+    ```
+   
+## Contract Overview
+
+### Why Attachments?
+
+- Why are we defining attachments instead of simply adding a field on the KittyVerse NFTs that can store hats?
+    - If we didn't define KittyVerse, we couldn't do that - no way we can alter the underlying contract
+        - Attachments allow us to permissionlessly extend existing resources with new data and functionality
+        - Why would we do that anyway?
+            
+            - If KittyVerse already has an enthusiastic community behind it
+    - Let's say we're defining both, why use attachments?
+        - Building with attachments is a clear signal that you're designing for this permissionless composability
+            - You encourage others to define their own sorts of attachments for KittyVerse NFTs
```

Comment:
is it possible to create attachments with changing the underlying contract ?

So let's say Alice creates a KittyVerse contract and it doesn't have any attachment functionality and bob owns one kittyVerse and wants to attach a kittyHat NFT to its KittyVerse NFT. Then bob can permisionlessely attach the kittyHat to its kittyVerse without the intervene of Alice changing the contract to add the support of attachments ?

Reply:
> is it possible to create attachments with changing the underlying contract ?
...
> bob can permisionlessely attach the kittyHat to its kittyVerse without the intervene of Alice changing the contract to add the support of attachments ?

Yes, that's correct. Bob would just create and deploy the KittyHats contract, and then he could add the KittyHats HatAttachment to his KittyVerse NFTs. No change needed to KittyVerse NFT